### PR TITLE
[chore] Update GCP codeowners

### PR DESCRIPTION
**Pull request by damemi**
Updating CODEOWNERS entries for GCP exporters

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
cc @punya

**Review by atoulme — CHANGES_REQUESTED** [review]
@psx95 is unfortunately not listed as a member of OpenTelemetry. Codeowners must be members of OpenTelemetry to be listed, please see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20868

**Review by mx-psi — COMMENTED** [review]
To unblock this, we could leave this PR just for removing @tbarker25 and leave adding @psx95 for a future PR

**Comment by damemi** [comment]
@mx-psi I believe @psx95 is working on sponsorship for otel org membership

**Comment by damemi** [comment]
@psx95 is now a member of the OpenTelemetry org 🎉 

Also, updating this PR to move @tbarker25 to emeritus

**Review by atoulme — APPROVED** [review]

File: .github/CODEOWNERS
Comment (dmitryax) [inline .github/CODEOWNERS:56]: Do we want to start adding the emeritus comment? This needs to be a policy we agree on with @open-telemetry/collector-contrib-approvers. We haven't done that so far.

Comment (atoulme) [reply .github/CODEOWNERS:56]: I approved this PR because I was requesting changes earlier, but this indeed made me tick a bit. I don't see value in listing folks as emeritus, but maybe it's worth celebrating them elsewhere on README.md.

Comment (TylerHelmuth) [reply .github/CODEOWNERS:56]: I don't think we should list emeritus in this file which we use as an up-to-date source of truth for our CI automation

Comment (dmitryax) [reply .github/CODEOWNERS:56]: I agree. @damemi, can you please remove the comment?

Comment (jsuereth) [reply .github/CODEOWNERS:56]: FYI - https://github.com/open-telemetry/community/blob/main/community-membership.md#emeritus-maintainerapprovertriager

Comment (mx-psi) [reply .github/CODEOWNERS:56]: > FYI - https://github.com/open-telemetry/community/blob/main/community-membership.md#emeritus-maintainerapprovertriager

I don't think code owners as used in the Collector quite fit into the maintainer/approver/triager roles, but still I think it's a good idea to have both a current owners and an emeritus list on each component's README. 

However, I think we should deal with this separately: we are not doing it now for any other component, and we definitely don't want to do it in the CODEOWNERS file, so for this PR I agree with @dmitryax and @TylerHelmuth that we should remove the comment.

Comment (damemi) [reply .github/CODEOWNERS]: In the interest of keeping this scoped to the main goal, I updated this PR to just add @psx95 and not touch @tbarker25. The topic of emeritus approvers is broader than what I intended for this change. I'm sorry for the confusion, everyone.

**Comment by mx-psi** [comment]
Filed #23411 for the emeritus question

**Pull request merged**
